crates/store/re_data_source/src/data_source.rs

```diff
@@ -247,8 +253,17 @@ impl DataSource {
 
             #[cfg(feature = "grpc")]
             Self::RerunGrpcUrl { url } => {
+                let url =
+                    url::Url::parse(&url).with_context(|| format!("Invalid gRPC URL: {url}"))?;
                 re_grpc_client::stream_recording(url, on_msg).map_err(|err| err.into())
             }
+
+            #[cfg(feature = "grpc")]
+            Self::RerunGrpcCatalog { url } => {
+                let url =
+                    url::Url::parse(&url).with_context(|| format!("Invalid gRPC URL: {url}"))?;
+                re_grpc_client::stream_catalog(url, on_msg).map_err(|err| err.into())
```

Comment:
Again, rather than duplicating, we could have just renamed `stream_recording` -> `stream_redap`